pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,378 @@
+            if (delta > 0) {
+                const pool = client.stratis_pools[stratis_bdev.Pool];
+                return pool.GrowPhysicalDevice(stratis_bdev.Uuid).then(std_reply);
+            } else if (delta < 0)
+                return Promise.reject(_("Stratis blockdevs can not be made smaller"));
+            else
+                return Promise.resolve();
+        } else if (size < orig_size) {
+            // This shouldn't happen.  But if it does, continuing is harmful, so we throw an error.
+            return Promise.reject(_("Unrecognized data can not be made smaller here."));
+        } else {
+            // Growing unrecognized content, nothing to do.
+            return Promise.resolve();
+        }
+    }
+
+    function crypto_resize() {
+        if (crypto) {
+            const opts = { };
+            if (passphrase)
+                opts.passphrase = { t: "s", v: passphrase };
+            return crypto.Resize(size - crypto_overhead, opts);
+        } else {
+            return Promise.resolve();
+        }
+    }
+
+    function lvm_resize() {
+        if (size != lvol.Size)
+            return lvol.Resize(size, { });
+        else
+            return Promise.resolve();
+    }
+
+    if (fsys && !fsys.Resize) {
+        // Fallback for old versions of UDisks.  This doesn't handle encrypted volumes.
+        if (size != orig_size) {
+            return lvol.Resize(size, { resize_fsys: { t: 'b', v: true } });
+        }
+    } else {
+        if (size < orig_size) {
+            return fsys_resize().then(crypto_resize)
+                    .then(lvm_resize);
+        } else if (size >= orig_size) {
+            return lvm_resize().then(crypto_resize)
+                    .then(fsys_resize);
+        }
+    }
+}
+
+export function get_resize_info(client, block, to_fit) {
+    let info, shrink_excuse, grow_excuse;
+
+    if (block) {
+        if (block.IdUsage == 'crypto' && client.blocks_crypto[block.path]) {
+            const encrypted = client.blocks_crypto[block.path];
+            const cleartext = client.blocks_cleartext[block.path];
+
+            if (!encrypted.Resize) {
+                info = { };
+                shrink_excuse = grow_excuse = _("Encrypted volumes can not be resized here.");
+            } else if (!cleartext) {
+                info = { };
+                shrink_excuse = grow_excuse = _("Encrypted volumes need to be unlocked before they can be resized.");
+            } else {
+                return get_resize_info(client, cleartext, to_fit);
+            }
+        } else if (block.IdUsage == 'filesystem') {
+            info = client.fsys_info && client.fsys_info[block.IdType];
+
+            if (!info) {
+                info = { };
+                shrink_excuse = grow_excuse = cockpit.format(_("$0 filesystems can not be resized here."),
+                                                             block.IdType);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-061737-0f5b7102-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Reply:
Unneeded fallback

Reply:
No, this is a untested feature.

Reply:
This is hard to test, UDisks2 has info for everything. But this might happen. I'll mark it as not-covered, since this isn't f38 specific.

Reply:
No! Let's fake a non-existing filesystem type via udev.